### ISSUE TITLE
Upgrading to xunit 2.3.0-rc3 causes projects to stop producing nupkg when calling dotnet-pack

### ISSUE THREAD

**Issue by natemcmaster** [xunit/xunit#1487]
In xunit 2.3.0-rc3, the `IsTestProject` MSBuild property is automatically being set to `true`. NuGet's packing targets disable `/t:Pack` when `IsTestProject` is true. (see [here](https://github.com/NuGet/NuGet.Client/blob/cd5525a54cdd8738f6c741b560c30d91a7f0ad1e/src/NuGet.Core/NuGet.Build.Tasks.Pack/Pack.targets#L30)). This means upgrading from 2.2.0 to 2.3.0 make cause nupkg's to stop being produced.

**Repro**
1. make a project
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="xunit.core" Version="2.2.0" />
  </ItemGroup>
</Project>
```
2. Produce nupkg
```
PS> dotnet pack
  xunit-ext -> C:\tmp\xunit-ext\bin\Debug\netstandard2.0\xunit-ext.dll
  Successfully created package 'C:\tmp\xunit-ext\bin\Debug\xunit-ext.1.0.0.nupkg'.
```
3. Upgrade to 2.3.0-rc3.
```xml
    <PackageReference Include="xunit.core" Version="2.3.0-rc3-build3818" />
```
4. `rm -r bin/` && `dotnet pack`

**Result:**
No package is produced when xunit.core/2.3.0-rc3 is referenced.

**Impact**
Only affects devs who produce nuget packages that depend on xunit, such as xunit extensions, plugins, etc.

**Workaround**
Devs need to add an explicit IsPackable property.
```xml
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <IsPackable>true</IsPackable>
  </PropertyGroup>
```

**Comment by bradwilson** [comment]
Out of curiosity, why would you pack a unit test project?

**Comment by bradwilson** [comment]
> Devs need to add an explicit IsPackable property.

The default template puts this property in the project file, though it's set to `false`.

**Comment by natemcmaster** [comment]
IsPackable=false is the default in `dotnet new xunit`, but not `dotnet new classlib`.

Example scenario: https://github.com/aspnet/Testing/tree/dev/src/Microsoft.AspNetCore.Testing

This project contains extension to xunit, such as custom FactAttribute implementations, but doesn't actually contain tests.

Other examples:
https://github.com/aspnet/DependencyInjection/tree/dev/src/DI.Specification.Tests
https://github.com/aspnet/EntityFrameworkCore/tree/dev/src/EFCore.Specification.Tests

We ship unit test "specs" to make it easier for third-parties to check their implementations of our abstractions.

**Comment by natemcmaster** [comment]
I'm not sure if either of these scenarios are super common, so I would be okay if this was just considered a "known issue". The workaround is super easy -- just add the new property -- but it's hard to discover if you don't know the black magic of MSBuild.

**Comment by bradwilson** [comment]
I'll put this in the 2.3 RTM release notes. Thanks!

**Comment by gulbanana** [comment]
what's the best way to override this for a library which extends xunit - `<IsTestProject>False</IsTestProject>`?

**Comment by bradwilson** [comment]
Set `<IsPackable>true</IsPackable>`.

But also, this release note:

> As a reminder: If you're extending xUnit.net and want to publish your extension as a NuGet package, you should import xunit.extensibility.core and/or xunit.extensibility.execution, not xunit or xunit.core. If you do this wrong, you might have problems generating your NuGet package via dotnet pack.

Make sure you're actually consuming the correct packages in the first place, and the problem should disappear. :)

**Comment by gulbanana** [comment]
xunit.extensibility.execution it is, thanks

**Comment by abdulbeard** [comment]
I spent all of today trying to figure out why my project wouldn't pack. Came down to manually editing the csproj, until I found the xunit reference. When included in the references, the project wouldn't pack. When excluded, it would. So I googled and found this thread.
Based on @bradwilson 's recommendation, switched out nuget package from xunit to xunit.extensibility.core. Worked like a charm.
Many hours wasted. But to some avail.
